src/otaclient/ota_core.py

```diff
@@ -536,40 +545,45 @@ def _execute_update(self):
 
         # ------ pre-update ------ #
         logger.info("enter local OTA update...")
+
+        with TemporaryDirectory() as _tmp_dir:
+            use_inplace_mode = can_use_in_place_mode(
+                dev=self._boot_controller.standby_slot_dev,
+                mnt_point=_tmp_dir,
+                threshold_in_bytes=int(
+                    self._ota_metadata.total_regulars_size
+                    * STANDBY_SLOT_USED_SIZE_THRESHOLD
+                ),
+            )
+        logger.info(
+            f"check if we can use in-place mode to update standby slot: {use_inplace_mode}"
+        )
+
         self._boot_controller.pre_update(
             self.update_version,
-            standby_as_ref=False,  # NOTE: this option is deprecated and not used by bootcontroller
-            erase_standby=True,  # NOTE: as of now, we only have rebuild mode, so always erase_standby
+            # NOTE: this option is deprecated and not used by bootcontroller
+            # TODO:(20250613) when standby_as_ref is set, skip mounting active slot.
+            #       we cannot do this for now, as some boot controller impl still refer to
+            #       active_slot mounts.
+            standby_as_ref=use_inplace_mode,
+            erase_standby=not use_inplace_mode,
```

Comment:
Ideally, when we use inplace mode, we can set `standby_as_ref=True` to skip mounting active slot, but grub boot controller and jetson boot controller will look for boot config files in active slot.
So the `standby_as_ref` option is still not used.